### PR TITLE
Create Application from IS owned Templates

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add modal for registering applications from IS templates

**Adding sample application templates**

1) Run `components/director/run.sh` in Compass directory.
2) Enter `https://compass-gateway.kyma.local/director/`, supply tenant and auth token headers.
3) Create a template by query:
```
mutation add_template {
  createApplicationTemplate( in: {
      name: "app-template"
      description: "app-template-desc"
      applicationInput: {
        name: "app"
        providerName: "compass"
        description: "test {{new-placeholder}}"
        labels: { a: ["b", "c"], d: ["e", "f"] }
        webhooks: [{ type: CONFIGURATION_CHANGED, url: "http://url.com" }]
        healthCheckURL: "http://url.valid"
      }
      placeholders: [
        { name: "new-placeholder", description: "new-placeholder-desc" }
      ]
      accessLevel: GLOBAL
    }) {
    id
  }
}
```

**Related issue(s)**
[Kyma](https://github.com/kyma-project/kyma/pull/6818)
[Compass](https://github.com/kyma-incubator/compass/pull/621)

### CODE REVIEW

**Comment by kwiatekus** [comment]
Be aware that templates providers would need to define placeholder to be used as application name.
Otherwise it would not be possible to create two applications out of one template:

```
mutation add_template {
  createApplicationTemplate( in: {
      name: "app-template2"
      description: "app-template-desc"
      applicationInput: {
        name: "{{app-name}}"
        providerName: "compass"
        description: "test {{new-placeholder}}"
        labels: { a: ["b", "c"], d: ["e", "f"] }
        webhooks: [{ type: CONFIGURATION_CHANGED, url: "http://url.com" }]
        healthCheckURL: "http://url.valid"
      }
      placeholders: [
        { name: "new-placeholder", description: "new-placeholder-desc" },
        { name: "app-name", description: "Application name" }
      ]
      accessLevel: GLOBAL
    }) {
    id
  }
}
```

**Comment by Wawrzyn321** [comment]
Follow-up with placeholder regex [1513](https://github.com/kyma-project/console/pull/1513).

File: compass/src/components/Applications/CreateApplicationFromTemplateModal/CreateApplicationFromTemplateModal.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        )}
+      </Menu>
+    );
+  };
+
+  const dropdownControlText = () => {
+    if (template) {
+      return template.name;
+    } else if (templatesQuery.error) {
+      console.warn(templatesQuery.error);
```
Comment (dariadomagala) [inline compass/src/components/Applications/CreateApplicationFromTemplateModal/CreateApplicationFromTemplateModal.js]: I think we should also show this error somewhere.
But most important is.. it doesn't work. It's shown for a second, then changes to 'Choose template' and if I click it, the app crashes:
Screen Shot 2020-01-17 at 15 16 10

File: compass/src/components/Applications/CreateApplicationFromTemplateModal/CreateApplicationFromTemplateModal.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      });
+      applicationsQuery.refetch();
+    } catch (e) {
+      console.warn(e);
+      let message = e.message;
+      if (e.message.match('Object is not unique')) {
+        message =
+          'Application with that name already exists, please choose another template.';
+      }
+      LuigiClient.uxManager().showAlert({
```
Comment (dariadomagala) [inline compass/src/components/Applications/CreateApplicationFromTemplateModal/CreateApplicationFromTemplateModal.js]: We could use our notifications here.

File: compass/src/components/Applications/Applications.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.create-application-dropdown {
+  li {
+    list-style-type: none;
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply compass/src/components/Applications/Applications.scss]: Good point. Set for all `li`s that are a descendant of `Popover`.

**Pull request merged**